src/helpers.rs

```diff
@@ -955,7 +955,6 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
 /// The result of that search is cached so that later calls are approximately free.
 #[derive(Clone)]
 pub struct CurrentSpan<'a, 'mir, 'tcx> {
-    current_frame_idx: Option<usize>,
     machine: &'a MiriMachine<'mir, 'tcx>,
 }
```

Comment:
Well we can't just pass a `Span` around, because we do want the functionality of `CurrentSpan::get_caller`, on some retags. So yes I agree it is now a bit of a silly type (and I need to fix up its docs) but I am not sure the code is improved by removing it.

Reply:
IMO it'd make more sense to pass around a machine -- that might also make some other arguments redundant then.